### PR TITLE
arduino-cli: 0.12.1 -> 0.17.0

### PR DESCRIPTION
**Pull request by freezeboy**
###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/arduino/arduino-cli/default.nix [hunk 1]
```diff
@@ -4,18 +4,18 @@ let
...
-      sha256 = "1jlxs4szss2250zp8rz4bislgnzvqhxyp6z48dhx7zaam03hyf0w";
+      sha256 = "7p86mhBg0U8cBZR1P43eBWS9fbMBJ9l5ga0IOedNzm4=";
     };
 
     subPackages = [ "." ];
 
-    vendorSha256 = "03yj2iar63qm10fw3jh9fvz57c2sqcmngb0mj5jkhbnwf8nl7mhc";
+    vendorSha256 = "kPIhG6lsH+0IrYfdlzdv/X/cUQb22Xza9Q6ywjKae/4=";
 
     doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/arduino/arduino-cli/default.nix:20]: Can you look if we could enable tests?

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 117492` at e8ab2357 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>arduino-ci</li>
    <li>arduino-cli</li>
  </ul>
</details>

**Comment by freezeboy** [comment]
@SuperSandro2000 Good idea, and it looks like they work :tada:

**Comment by petabyteboy** [comment]
The update has been done in https://github.com/NixOS/nixpkgs/pull/122176
Do you want to re-purpose this PR for enabling the tests?

**Pull request closed without merging**